distributed/protocol/numpy.py

```diff
@@ -33,22 +32,36 @@ def serialize_numpy_ndarray(x):
         frames = [pickle.dumps(x)]
         return header, frames
 
+    # We cannot blindly pickle the dtype as some may fail pickling,
+    # so we have a mixture of strategies.
     if x.dtype.kind == 'V':
-        dt = x.dtype.descr
+        # Preserving all the information works best when pickling
+        try:
+            # Only use stdlib pickle as cloudpickle is slow when failing
+            # (microseconds instead of nanoseconds)
+            dt = (1, pickle.pickle.dumps(x.dtype))
+            pickle.loads(dt[1])  # does it unpickle fine?
+        except Exception:
+            # dtype fails pickling => fall back on the descr if reasonable.
+            if x.dtype.type is not np.void or x.dtype.alignment != 1:
+                raise
+            else:
+                dt = (0, x.dtype.descr)
     else:
-        dt = x.dtype.str
+        dt = (0, x.dtype.str)
```

Comment:
Yes, it's a boolean. I don't think it matters much but we can switch to a boolean if you prefer.